External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp

```diff
@@ -188,6 +202,141 @@ uint64_t AtomicFetchNeg(uint64_t *Addr) {
   return Result;
 }
 
+template<typename T>
+static T AtomicCompareAndSwap(T expected, T desired, T *addr);
+
+template<>
+uint8_t AtomicCompareAndSwap(uint8_t expected, uint8_t desired, uint8_t *addr) {
```

Comment:
Technically this drops ARMv8.1+ builds to using ARMv8.0 operations, but since it is the interpreter it isn't a big deal.
I'll let this go for now and create an issue to make it more optimal in the future, when we improve the dispatch logic of the interpreter.